lib/logstash/outputs/webhdfs.rb

```diff
@@ -0,0 +1,282 @@
+
+  config_name "webhdfs"
+  milestone 1
+
+  if RUBY_VERSION[0..2] == '1.8'
+    MAGIC = "\x82SNAPPY\x0"
+  else
+    MAGIC = "\x82SNAPPY\x0".force_encoding Encoding::ASCII_8BIT
+  end
+  DEFAULT_VERSION = 1
+  MINIMUM_COMPATIBLE_VERSION = 1
+
+  # The server name and port for webhdfs/httpfs connections.
+  config :server, :validate => :string, :required => true
+
+  # The Username for webhdfs.
+  config :user, :validate => :string, :required => false
+
+  # The path to the file to write to. Event fields can be used here,
+  # as well as date fields in the joda time format, e.g.:
+  # ....
+  #     "/user/logstash/dt=%{+YYYY-MM-dd}/%{@source_host}-%{+HH}.log"
+  # ....
+  config :path, :validate => :string, :required => true
+
+  # The format to use when writing events to the file. This value
+  # supports any string and can include %{name} and other dynamic
+  # strings.
+  #
+  # If this setting is omitted, the full json representation of the
+  # event will be written as a single line.
+  config :message_format, :validate => :string
+
+  # Sending data to webhdfs in x seconds intervals.
+  config :idle_flush_time, :validate => :number, :default => 1
+
+  # Sending data to webhdfs if event count is above, even if store_interval_in_secs is not reached.
+  config :flush_size, :validate => :number, :default => 500
+
+  # WebHdfs open timeout, default 30s (in ruby net/http).
+  config :open_timeout, :validate => :number, :default => 30
+
+  # The WebHdfs read timeout, default 30s (in ruby net/http).
+  config :read_timeout, :validate => :number, :default => 30
+
+  # Use httpfs mode if set to true, else webhdfs.
+  config :use_httpfs, :validate => :boolean, :default => false
+
+  # Retry some known webhdfs errors. These may be caused by race conditions when appending to same file, etc.
+  config :retry_known_errors, :validate => :boolean, :default => true
+
+  # How long should we wait between retries.
+  config :retry_interval, :validate => :number, :default => 0.5
+
+  # How many times should we retry.
+  config :retry_times, :validate => :number, :default => 5
+
+  # Compress output. One of ['none', 'snappy', 'gzip']
+  config :compression, :validate => ["none", "snappy", "gzip"], :default => "none"
+
+  # Set snappy chunksize. Only neccessary for stream format. Defaults to 32k. Max is 65536
+  # @see http://code.google.com/p/snappy/source/browse/trunk/framing_format.txt
+  config :snappy_bufsize, :validate => :number, :default => 32768
+
+  # Set snappy format. One of "stream", "file". Set to stream to be hive compatible.
+  config :snappy_format, :validate => ["stream", "file"], :default => "stream"
+
+  # Remove @timestamp field. Hive does not like a leading "@", but we need @timestamp for path calculation.
+  config :remove_at_timestamp, :validate => :boolean, :default => true
+
+  public
+
+  def register
+    begin
+      require 'webhdfs'
+    rescue LoadError
+      @logger.error("Module webhdfs could not be loaded.")
```

Comment:
This apply to the other require too, can the plugin work if some of this modules are not available? The way this is done here it would catch the error, report it, but got the error masqueraded.